packages/@aws-cdk/aws-chatbot/test/slack-channel-configuration.test.ts

```diff
@@ -0,0 +1,361 @@
+          },
+        ],
+        Version: '2012-10-17',
+      },
+      Description: 'LambdaInvoke policy for AWS Chatbot',
+      ManagedPolicyName: 'AWS-Chatbot-LambdaInvoke-Policy',
+      Path: '/',
+    }));
+  });
+
+  test('new configuration with new role and add support command permissions', () => {
+    const stack = new cdk.Stack();
+
+    const slackChannel = new chatbot.SlackChannelConfiguration(stack, 'MySlackChannel', {
+      slackWorkspaceId: 'ABC123',
+      slackChannelId: 'DEF456',
+      slackChannelConfigurationName: 'Test',
+    });
+
+    slackChannel.addSupportCommandPermissions();
+
+    cdkAssert.expect(stack).to(cdkAssert.haveResource('AWS::IAM::Role', {
+      AssumeRolePolicyDocument: {
+        Statement: [
+          {
+            Action: 'sts:AssumeRole',
+            Effect: 'Allow',
+            Principal: {
+              Service: 'chatbot.amazonaws.com',
+            },
+          },
+        ],
+        Version: '2012-10-17',
+      },
+      ManagedPolicyArns: [
+        {
+          'Fn::Join': [
+            '',
+            [
+              'arn:',
+              {
+                Ref: 'AWS::Partition',
+              },
+              ':iam::aws:policy/AWSSupportAccess',
+            ],
+          ],
+        },
+      ],
+    }));
+  });
+
+  test('new configuration with new role and extra iam policies', () => {
+    const stack = new cdk.Stack();
+
+    const slackChannel = new chatbot.SlackChannelConfiguration(stack, 'MySlackChannel', {
+      slackWorkspaceId: 'ABC123',
+      slackChannelId: 'DEF456',
+      slackChannelConfigurationName: 'Test',
+    });
+
+    slackChannel.addToPrincipalPolicy(new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        's3:GetObject',
+      ],
+      resources: ['arn:aws:s3:::abc/xyz/123.txt'],
+    }));
+
+    cdkAssert.expect(stack).to(cdkAssert.haveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 's3:GetObject',
+            Effect: 'Allow',
+            Resource: 'arn:aws:s3:::abc/xyz/123.txt',
+          },
+        ],
+        Version: '2012-10-17',
+      },
+    }));
+  });
+
+  test('added a iam policy to a from slack channel configuration ARN will nothing to do', () => {
+    const stack = new cdk.Stack();
+    const imported = chatbot.SlackChannelConfiguration.fromSlackChannelConfigurationArn(stack, 'MySlackChannel', 'arn:aws:chatbot::1234567890:chat-configuration/slack-channel/my-slack');
+
+    (imported as chatbot.SlackChannelConfiguration).addToPrincipalPolicy(new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        's3:GetObject',
+      ],
+      resources: ['arn:aws:s3:::abc/xyz/123.txt'],
+    }));
+
+    cdkAssert.expect(stack).to(cdkAssert.countResources('AWS::IAM::Role', 0));
+    cdkAssert.expect(stack).to(cdkAssert.countResources('AWS::IAM::Policy', 0));
+  });
+
+  test('from slack channel configuration ARN', () => {
```

Comment:
There are no tests for imported SlackChannelConfiguration with addXPermissions.

Reply:
+1!